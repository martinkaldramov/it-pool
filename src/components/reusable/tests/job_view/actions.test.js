import React from 'react';
import { shallow } from 'enzyme';
import Actions from '../../job_view/actions';

describe('Job Actions', () => {
  const mockApply = jest.fn();
  const mockSave = jest.fn();
  const wrapper = shallow(<Actions apply={mockApply} save={mockSave} />);

  it('should render successfully', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call the saveJob function when the SAVE JOB button is clicked', () => {
    wrapper.find('.job-apply').simulate('click');
    expect(mockApply).toHaveBeenCalled();
  });

  it('should call the apply function when the APPLY button is clicked', () => {
    wrapper.find('.job-save').simulate('click');
    expect(mockSave).toHaveBeenCalled();
  });
});
