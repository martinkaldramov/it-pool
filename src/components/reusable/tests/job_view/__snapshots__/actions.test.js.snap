// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job Actions should render successfully 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Actions
    apply={[Function]}
    save={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <RaisedButton
          className="job-apply"
          disabled={false}
          fullWidth={false}
          label="APPLY"
          labelPosition="after"
          onClick={[Function]}
          primary={true}
          secondary={false}
/>,
        <RaisedButton
          className="job-save"
          disabled={false}
          fullWidth={false}
          label="SAVE JOB"
          labelPosition="after"
          onClick={[Function]}
          primary={true}
          secondary={false}
/>,
        <ContentFlag />,
        " Flag as inappropriate",
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "job-apply",
          "disabled": false,
          "fullWidth": false,
          "label": "APPLY",
          "labelPosition": "after",
          "onClick": [Function],
          "primary": true,
          "secondary": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "job-save",
          "disabled": false,
          "fullWidth": false,
          "label": "SAVE JOB",
          "labelPosition": "after",
          "onClick": [Function],
          "primary": true,
          "secondary": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      " Flag as inappropriate",
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <RaisedButton
            className="job-apply"
            disabled={false}
            fullWidth={false}
            label="APPLY"
            labelPosition="after"
            onClick={[Function]}
            primary={true}
            secondary={false}
/>,
          <RaisedButton
            className="job-save"
            disabled={false}
            fullWidth={false}
            label="SAVE JOB"
            labelPosition="after"
            onClick={[Function]}
            primary={true}
            secondary={false}
/>,
          <ContentFlag />,
          " Flag as inappropriate",
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "job-apply",
            "disabled": false,
            "fullWidth": false,
            "label": "APPLY",
            "labelPosition": "after",
            "onClick": [Function],
            "primary": true,
            "secondary": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "job-save",
            "disabled": false,
            "fullWidth": false,
            "label": "SAVE JOB",
            "labelPosition": "after",
            "onClick": [Function],
            "primary": true,
            "secondary": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        " Flag as inappropriate",
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
